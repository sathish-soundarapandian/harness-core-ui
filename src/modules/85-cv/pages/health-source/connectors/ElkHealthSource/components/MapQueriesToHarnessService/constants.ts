/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MapElkQueryToService } from './ElkQueryBuilder.types'

export const MapElkToServiceFieldNames = {
  METRIC_NAME: 'metricName',
  QUERY: 'query',
  TIMESTAMP_FORMAT: 'timeStampFormat',
  SERVICE_INSTANCE: 'serviceInstance',
  LOG_INDEXES: 'logIndexes',
  IDENTIFY_TIMESTAMP: 'identifyTimestamp',
  MESSAGE_IDENTIFIER: 'messageIdentifier',
  IS_STALE_RECORD: 'isStaleRecord'
}

export const initialFormData: MapElkQueryToService = {
  metricName: 'ELK Logs Query',
  query: '',
  timeStampFormat: '',
  serviceInstance: '',
  logIndexes: '',
  identifyTimestamp: '',
  messageIdentifier: ''
}
