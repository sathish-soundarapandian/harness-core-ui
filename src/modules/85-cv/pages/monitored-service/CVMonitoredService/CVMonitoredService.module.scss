/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.healthScoreCardContainer {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding-left: var(--spacing-large);
}

.healthScoreCard {
  height: 31px;
  min-width: 34px;
  border: none !important;
  font-size: var(--font-size-small);
  text-align: center;
  font-weight: bold;
}

.tags {
  p {
    color: var(--grey-600) !important;
    font-size: var(--font-size-small) !important;
  }
}

.tagsText {
  text-transform: capitalize;
}

.graphContainer {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.filterCard {
  margin-right: var(--spacing-small);
  margin-bottom: var(--spacing-small);
  padding: var(--spacing-small) var(--spacing-large) !important;
}

.filterSelect {
  margin-right: var(--spacing-large);
}

$headerHeight: 146px;

.pageContainer {
  height: calc(100% - #{$headerHeight});
  overflow: auto;
  position: relative;
}

.pageBody {
  --page-header-height: #{$headerHeight} !important;
}

.noDataContainer {
  height: 100% !important;
}

.infrastructureIconPadding {
  margin-left: -3px !important ;
}

.noServiceAvailableImage {
  width: 720px;
  height: 350px;
  margin-right: 40px;
}

.monitoredServiceName {
  white-space: nowrap !important;
  overflow: hidden !important;
  text-overflow: ellipsis !important;
}

.toggleFlagButton,
.toggleFlagButton:hover,
.toggleFlagButton:active {
  border-radius: var(--spacing-small);
  padding: 0 !important;
  border: none !important;
  background-color: transparent;

  :global {
    .bp3-control.bp3-switch input:checked ~ .bp3-control-indicator {
      background-color: var(--primary-4);
    }
  }
}

.monitoredServiceLink {
  overflow: hidden;
}
