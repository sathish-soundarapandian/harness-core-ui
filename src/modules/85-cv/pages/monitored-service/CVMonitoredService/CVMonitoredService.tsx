/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useState } from 'react'
import { useHistory, useParams } from 'react-router-dom'
import { defaultTo } from 'lodash-es'
import {
  Page,
  ButtonVariation,
  Layout,
  Select,
  GridListToggle,
  Views,
  SelectOption,
  Heading,
  HarnessDocTooltip,
  ExpandingSearchInput,
  Container
} from '@wings-software/uicore'
import { FontVariation } from '@harness/design-system'
import { useStrings } from 'framework/strings'
import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import { ResourceType } from '@rbac/interfaces/ResourceType'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import { useGetMonitoredServiceListEnvironments, useGetCountOfServices } from 'services/cv'
import { useFeatureFlags } from '@common/hooks/useFeatureFlag'
import routes from '@common/RouteDefinitions'
import { useQueryParams } from '@common/hooks'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import RbacButton from '@rbac/components/Button/Button'
import { useDocumentTitle } from '@common/hooks/useDocumentTitle'
import { TemplateUsage } from '@templates-library/utils/templatesUtils'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { useTemplateSelector } from 'framework/Templates/TemplateSelectorContext/useTemplateSelector'
import { getCVMonitoringServicesSearchParam, getErrorMessage, getEnvironmentOptions } from '@cv/utils/CommonUtils'
import ServiceDependencyGraph from '@cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceGraphView/MonitoredServiceGraphView'
import { getEnvironmentIdentifier } from './CVMonitoredService.utils'
import { FilterTypes } from './CVMonitoredService.types'
import MonitoredServiceList from './components/MonitoredServiceListView/MonitoredServiceList'
import css from './CVMonitoredService.module.scss'

const MonitoredService: React.FC = () => {
  const { getString } = useStrings()

  useDocumentTitle([getString('cv.srmTitle'), getString('cv.monitoredServices.title')])

  const history = useHistory()
  const { view } = useQueryParams<{ view?: Views.GRID }>()
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
  const pathParams = {
    accountId,
    orgIdentifier,
    projectIdentifier
  }
  const { CVNG_TEMPLATE_MONITORED_SERVICE } = useFeatureFlags()

  const [page, setPage] = useState(0)
  const [selectedView, setSelectedView] = useState<Views>(view === Views.GRID ? Views.GRID : Views.LIST)
  const [environment, setEnvironment] = useState<SelectOption>()
  const [search, setSearch] = useState<string>('')
  const [selectedFilter, setSelectedFilter] = useState<FilterTypes>(FilterTypes.ALL)

  useEffect(() => {
    setPage(0)
  }, [projectIdentifier])

  const { data: environmentDataList, loading: loadingEnvironments } = useGetMonitoredServiceListEnvironments({
    queryParams: pathParams
  })

  const {
    data: serviceCountData,
    loading: serviceCountLoading,
    error: serviceCountError,
    refetch: refetchServiceCountData
  } = useGetCountOfServices({
    queryParams: {
      ...pathParams,
      environmentIdentifier: getEnvironmentIdentifier(environment)
    }
  })

  useEffect(() => {
    /* istanbul ignore else */ if (serviceCountData) {
      refetchServiceCountData()
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [selectedView])

  const onFilter = (type: FilterTypes): void => {
    /* istanbul ignore else */ if (type !== selectedFilter) {
      setPage(0)
      setSelectedFilter(type)
      refetchServiceCountData()
    }
  }

  const { getTemplate } = useTemplateSelector()

  const onUseTemplate = async () => {
    const { template } = await getTemplate({ templateType: 'MonitoredService', allowedUsages: [TemplateUsage.USE] })
    const templateRefData = {
      identifier: template?.identifier,
      accountId: template?.accountId,
      orgIdentifier: template?.orgIdentifier,
      projectIdentifier: template?.projectIdentifier,
      versionLabel: template?.versionLabel
    }
    history.push({
      pathname: routes.toCVMonitoringServicesInputSets(pathParams),
      search: getCVMonitoringServicesSearchParam({
        view: selectedView,
        templateRef: JSON.stringify(templateRefData)
      })
    })
  }

  const createButton = (hasMonitoredServices: boolean) => {
    {
      const LayoutOrientation = hasMonitoredServices ? Layout.Horizontal : Layout.Vertical
      return (
        <LayoutOrientation spacing="large">
          <RbacButton
            variation={ButtonVariation.PRIMARY}
            icon="plus"
            text={getString('cv.monitoredServices.newMonitoredServices')}
            onClick={() => {
              history.push({
                pathname: routes.toCVAddMonitoringServicesSetup(pathParams),
                search: getCVMonitoringServicesSearchParam({ view: selectedView })
              })
            }}
            permission={{
              permission: PermissionIdentifier.EDIT_MONITORED_SERVICE,
              resource: {
                resourceType: ResourceType.MONITOREDSERVICE,
                resourceIdentifier: projectIdentifier
              }
            }}
          />
          {hasMonitoredServices && CVNG_TEMPLATE_MONITORED_SERVICE && (
            <RbacButton
              text={getString('common.useTemplate')}
              variation={ButtonVariation.SECONDARY}
              icon="template-library"
              onClick={onUseTemplate}
              featuresProps={{
                featuresRequest: {
                  featureNames: [FeatureIdentifier.TEMPLATE_SERVICE]
                }
              }}
            />
          )}
        </LayoutOrientation>
      )
    }
  }

  return (
    <>
      <Page.Header
        breadcrumbs={<NGBreadcrumbs />}
        title={
          <Heading level={3} font={{ variation: FontVariation.H4 }}>
            {getString('cv.monitoredServices.title')}
            <HarnessDocTooltip tooltipId={'monitoredServicesTitle'} useStandAlone />
          </Heading>
        }
      />
      <Page.Header
        title={createButton(Boolean(serviceCountData?.allServicesCount))}
        toolbar={
          <Layout.Horizontal spacing="medium">
            <Container data-name="monitoredServiceSeachContainer">
              <ExpandingSearchInput
                width={250}
                alwaysExpanded
                throttle={500}
                onChange={setSearch}
                placeholder={getString('cv.monitoredServices.searchMonitoredServices')}
              />
            </Container>
            <Select
              value={{
                label: `${getString('environment')}: ${defaultTo(environment?.label, getString('all'))}`,
                value: defaultTo(environment?.value, getString('all'))
              }}
              defaultSelectedItem={{ label: getString('all'), value: getString('all') }}
              items={getEnvironmentOptions(environmentDataList, loadingEnvironments, getString)}
              onChange={item => {
                setPage(0)
                setEnvironment(item)
              }}
              className={css.filterSelect}
            />
            <GridListToggle
              initialSelectedView={selectedView}
              onViewToggle={setSelectedView}
              icons={{ left: 'graph' }}
            />
          </Layout.Horizontal>
        }
      />
      {selectedView === Views.LIST ? (
        <MonitoredServiceList
          page={page}
          search={search}
          setPage={setPage}
          environmentIdentifier={getEnvironmentIdentifier(environment)}
          createButton={createButton(Boolean(!serviceCountData?.allServicesCount))}
          selectedFilter={selectedFilter}
          onFilter={onFilter}
          serviceCountData={serviceCountData}
          serviceCountLoading={serviceCountLoading}
          serviceCountErrorMessage={getErrorMessage(serviceCountError)}
          refetchServiceCountData={refetchServiceCountData}
        />
      ) : (
        <ServiceDependencyGraph
          isPageView
          search={search}
          serviceCountData={serviceCountData}
          selectedFilter={selectedFilter}
          onFilter={onFilter}
          refetchServiceCountData={refetchServiceCountData}
          serviceCountLoading={serviceCountLoading}
          createButton={createButton(Boolean(!serviceCountData?.allServicesCount))}
          environmentIdentifier={getEnvironmentIdentifier(environment)}
          serviceCountErrorMessage={getErrorMessage(serviceCountError)}
        />
      )}
    </>
  )
}

export default MonitoredService
