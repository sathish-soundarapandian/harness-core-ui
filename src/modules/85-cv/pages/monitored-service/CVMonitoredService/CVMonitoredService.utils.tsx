/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ReactElement } from 'react'
import React from 'react'
import { isNull, isNumber } from 'lodash-es'
import Highcharts, { type PointOptionsObject } from 'highcharts'
import { Text, Layout, Tag, type SelectOption, type IconName } from '@harness/uicore'
import { FontVariation, Color } from '@harness/design-system'
import HighchartsReact from 'highcharts-react-official'
import type { Renderer, CellProps } from 'react-table'
import { useStrings, type UseStringsReturn } from 'framework/strings'
import type { CategoryCountDetails, CountServiceDTO, MonitoredServiceListItemDTO, RiskData } from 'services/cv'
import { getRiskColorValue, getRiskLabelStringId } from '@cv/utils/CommonUtils'
import ImageDeleteService from '@cv/assets/delete-service.svg'
import type { FilterCardItem } from '@cv/components/FilterCard/FilterCard.types'
import type { RiskTagWithLabelProps } from './CVMonitoredService.types'
import { FilterTypes } from './CVMonitoredService.types'
import { HistoricalTrendChartOption, DefaultChangePercentage } from './CVMonitoredService.constants'
import css from './CVMonitoredService.module.scss'

export const getEnvironmentIdentifier = (environment?: SelectOption): string | undefined => {
  if (environment?.value !== 'All') {
    return environment?.value as string
  }
}

export const createTrendDataWithZone = (trendData: RiskData[]): Highcharts.SeriesLineOptions => {
  const highchartsLineData: PointOptionsObject[] = []
  let currentRiskColor: string | null = getRiskColorValue(trendData?.[0].riskStatus)
  const zones: Highcharts.SeriesZonesOptionsObject[] = [{ value: undefined, color: currentRiskColor }]

  trendData.forEach((dataPoint, index) => {
    const { riskStatus } = dataPoint || {}
    let { healthScore } = dataPoint || {}
    if (isNull(healthScore)) {
      healthScore = -2
    }
    const riskColor = getRiskColorValue(riskStatus)
    highchartsLineData.push({ x: index, y: healthScore })
    if (isNumber(healthScore) && riskStatus) {
      if (riskColor !== currentRiskColor) {
        zones[zones.length - 1].value = index
        zones.push({ value: undefined, color: riskColor })
        currentRiskColor = riskColor
      }
    } else {
      currentRiskColor = null
    }
  })

  return {
    data: highchartsLineData,
    zones,
    name: '',
    type: 'line',
    zoneAxis: 'x',
    clip: false
  }
}

export const getHistoricalTrendChartOption = (trendData: RiskData[]): Highcharts.Options => {
  return {
    ...HistoricalTrendChartOption,
    series: [
      {
        ...createTrendDataWithZone(trendData)
      }
    ]
  }
}

export const ServiceHealthTrend = ({ healthScores }: { healthScores?: RiskData[] }): JSX.Element => {
  if (!healthScores) {
    return <></>
  }

  return <HighchartsReact highcharts={Highcharts} options={getHistoricalTrendChartOption(healthScores)} />
}

export const RenderHealthTrend: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }) => {
  const healthScores = row.original.historicalTrend?.healthScores

  return <ServiceHealthTrend healthScores={healthScores} />
}

export const RiskTagWithLabel = ({
  riskData,
  labelVariation,
  color,
  label,
  isDarkBackground
}: RiskTagWithLabelProps): ReactElement => {
  const { getString } = useStrings()
  const { riskStatus, healthScore } = riskData ?? {}

  return (
    <Layout.Horizontal className={css.healthScoreCardContainer} spacing="small">
      <Tag
        className={css.healthScoreCard}
        style={{ backgroundColor: getRiskColorValue(riskStatus, true, !!isDarkBackground) }}
      >
        {healthScore || healthScore === 0 ? healthScore : '-'}
      </Tag>
      <Text color={color ?? Color.BLACK} font={{ variation: labelVariation ?? FontVariation.BODY }}>
        {label ?? getString(getRiskLabelStringId(riskStatus))}
      </Text>
    </Layout.Horizontal>
  )
}

export const RenderHealthScore: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }) => {
  const monitoredService = row.original

  if (!monitoredService.healthMonitoringEnabled) {
    return <></>
  }

  return <RiskTagWithLabel riskData={monitoredService.currentHealthScore} />
}

export const calculateTotalChangePercentage = (
  changeSummaryTotal?: CategoryCountDetails
): { color: string; percentage: number; icon: IconName } => {
  if (changeSummaryTotal?.percentageChange) {
    const { percentageChange } = changeSummaryTotal
    return {
      color: percentageChange > 0 ? Color.SUCCESS : Color.ERROR,
      percentage: Math.abs(percentageChange),
      icon: percentageChange > 0 ? 'symbol-triangle-up' : 'symbol-triangle-down'
    }
  }
  return DefaultChangePercentage
}

export const ServiceDeleteContext = ({ serviceName }: { serviceName?: string }): ReactElement => {
  const { getString } = useStrings()

  return (
    <Layout.Horizontal flex={{ justifyContent: 'space-between', alignItems: 'flex-start' }}>
      <Text color={Color.GREY_800}>
        {getString('cv.monitoredServices.deleteMonitoredServiceWarning', { name: serviceName })}
      </Text>
      <div>
        <img src={ImageDeleteService} width="204px" height="202px" />
      </div>
    </Layout.Horizontal>
  )
}

export const getMonitoredServiceFilterOptions = (
  getString: UseStringsReturn['getString'],
  serviceCountData: CountServiceDTO | null
): FilterCardItem[] => {
  return [
    {
      type: FilterTypes.ALL,
      title: getString('common.allServices'),
      icon: 'services',
      iconSize: 18,
      count: serviceCountData?.allServicesCount
    },
    {
      type: FilterTypes.RISK,
      title: getString('cv.servicesAtRisk'),
      icon: 'offline-outline',
      iconSize: 18,
      count: serviceCountData?.servicesAtRiskCount
    }
  ]
}
