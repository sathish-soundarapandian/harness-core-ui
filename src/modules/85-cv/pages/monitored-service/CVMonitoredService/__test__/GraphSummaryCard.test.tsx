/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, queryByText, waitFor, screen } from '@testing-library/react'
import * as cvService from 'services/cv'
import routes from '@common/RouteDefinitions'
import { findDialogContainer, TestWrapper } from '@common/utils/testUtils'
import { getCVMonitoringServicesSearchParam } from '@cv/utils/CommonUtils'
import { monitoredService } from '@cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.mock'
import { SummaryCard } from '../components/MonitoredServiceGraphView/views/SummaryCard'
import {
  testWrapperProps,
  pathParams,
  errorMessage,
  servicePoint
} from '../components/MonitoredServiceGraphView/__tests__/ServiceDependencyGraph.mock'
import { MonitoredServiceEnum } from '../../MonitoredServicePage.constants'

const onToggleService = jest.fn()
const onDeleteService = jest.fn()

jest.mock('services/cv', () => ({
  useGetMonitoredServiceDetailsWithServiceId: jest
    .fn()
    .mockImplementation(() => ({ data: monitoredService, refetch: jest.fn() }))
}))

describe('Monitored Service Summary Card', () => {
  test('Service data available', () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <SummaryCard onToggleService={onToggleService} onDeleteService={onDeleteService} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('onEditService', async () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <SummaryCard onToggleService={onToggleService} onDeleteService={onDeleteService} />
      </TestWrapper>
    )

    await userEvent.click(container.querySelector('[data-icon="Edit"]')!)

    await waitFor(() =>
      expect(
        screen.getByText(
          routes.toCVAddMonitoringServicesEdit({ ...pathParams, identifier: 'identifier', module: 'cv' }) +
            getCVMonitoringServicesSearchParam({ tab: MonitoredServiceEnum.Configurations })
        )
      ).toBeInTheDocument()
    )
  })

  test('onDeleteService', async () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <SummaryCard onToggleService={onToggleService} onDeleteService={onDeleteService} />
      </TestWrapper>
    )

    await userEvent.click(container.querySelector('[data-icon="trash"]')!)

    await waitFor(() => screen.queryByText('yes'))

    const confirmDialog = findDialogContainer()

    expect(confirmDialog).toBeInTheDocument()

    await userEvent.click(queryByText(confirmDialog!, 'yes')!)

    expect(onDeleteService).toBeCalledWith(monitoredService.identifier)
  })

  test('onToggleService', async () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <SummaryCard onToggleService={onToggleService} onDeleteService={onDeleteService} />
      </TestWrapper>
    )

    await userEvent.click(container.querySelector('[data-name="on-btn"]')!)

    expect(onToggleService).toHaveBeenCalledWith(monitoredService.identifier, true)
  })

  test('Loading', () => {
    jest
      .spyOn(cvService, 'useGetMonitoredServiceDetailsWithServiceId')
      .mockReturnValue({ loading: true, refetch: jest.fn() } as any)

    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <SummaryCard onToggleService={onToggleService} onDeleteService={onDeleteService} />
      </TestWrapper>
    )

    expect(container.querySelector('[data-icon="steps-spinner"]')).toBeInTheDocument()
  })

  test('Error', async () => {
    jest
      .spyOn(cvService, 'useGetMonitoredServiceDetailsWithServiceId')
      .mockReturnValue({ error: { message: errorMessage }, refetch: jest.fn() } as any)

    render(
      <TestWrapper {...testWrapperProps}>
        <SummaryCard onToggleService={onToggleService} onDeleteService={onDeleteService} point={servicePoint} />
      </TestWrapper>
    )

    expect(screen.getByText(errorMessage)).toBeInTheDocument()

    await userEvent.click(screen.getByText('Retry'))

    await waitFor(() =>
      expect(cvService.useGetMonitoredServiceDetailsWithServiceId).toHaveBeenLastCalledWith({
        monitoredServiceIdentifier: servicePoint.monitoredServiceIdentifier,
        queryParams: pathParams
      })
    )
  })
})
