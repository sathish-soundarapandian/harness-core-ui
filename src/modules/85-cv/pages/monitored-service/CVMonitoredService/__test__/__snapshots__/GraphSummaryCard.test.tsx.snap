// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitored Service Summary Card Service data available 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--flex"
  >
    <div
      class="StyledProps--font StyledProps--main"
    >
      <a
        href="/account/account_id/cv/orgs/org_identifier/projects/project_identifier/monitoringservices/edit/identifier?tab=ServiceHealth"
      >
        <p
          class="StyledProps--font StyledProps--main serviceName StyledProps--color StyledProps--color-primary7 StyledProps--font-variation-h6"
        >
          serviceName
        </p>
      </a>
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-small"
      >
        environmentName
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
        type="button"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey0"
          data-icon="Edit"
        >
          <svg
            height="14"
            viewBox="0 0 16 16"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
              fill="currentColor"
              fill-rule="evenodd"
              opacity="0.3"
            />
          </svg>
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-right-small Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-trash StyledProps--main StyledProps--color StyledProps--color-grey0"
          icon="trash"
        >
          <svg
            data-icon="trash"
            height="14"
            viewBox="0 0 16 16"
            width="14"
          >
            <desc>
              trash
            </desc>
            <path
              d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <div
        class="optionBtn"
      >
        <div
          class="item active disabled"
          data-name="on-btn"
          role="button"
        >
          common.ON
        </div>
        <div
          class="item"
          data-name="off-btn"
          role="button"
        >
          common.OFF
        </div>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--border StyledProps--border-top StyledProps--border-bottom StyledProps--border-color-grey400 StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--margin-top-medium StyledProps--margin-bottom-medium"
  >
    <p
      class="StyledProps--font StyledProps--main serviceChanges StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-tiny-semi StyledProps--padding-bottom-medium"
    >
      cv.Dependency.serviceChanges
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey100"
        style="display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey100"
          data-icon="nav-project"
        >
          <svg
            height="12"
            viewBox="0 0 16 16"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.163 1.126a.5.5 0 00-.5 0L2.056 4.36a.5.5 0 00-.25.433v6.234a.5.5 0 00.248.432l5.87 3.415a.5.5 0 00.501 0l5.631-3.259a.5.5 0 00.25-.433V7.478a.5.5 0 00-1 0v3.417l-4.625 2.677.035-5.07 5.34-3.091a.5.5 0 000-.865l-5.894-3.42zM7.717 8.5l-.035 5.077-4.875-2.837V5.66l4.91 2.84zm.502-.865l-4.913-2.84 4.606-2.659 4.897 2.842-4.59 2.657zm-3.864-.068a.5.5 0 10-.5.866l2.165 1.25a.5.5 0 10.5-.866l-2.165-1.25z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        10 Deployments
      </p>
      <p
        class="StyledProps--font StyledProps--main infrastructureIconPadding StyledProps--font-variation-small StyledProps--color StyledProps--color-grey100"
        style="display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-none StyledProps--color StyledProps--color-grey100"
          data-icon="infrastructure"
        >
          <svg
            fill="currentColor"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.15 10.822l2.091-3.623-2.091-3.624H6.966L4.874 7.2l2.092 3.623h4.184zm2.944-4.106l-.004-.007-2.092-3.623a.98.98 0 00-.848-.49H6.966a.98.98 0 00-.848.49L4.026 6.709a.98.98 0 000 .98l2.092 3.623.007.012-.007.013-2.092 3.623a.98.98 0 000 .98l2.092 3.623a.98.98 0 00.848.49h4.184a.98.98 0 00.848-.49l2.092-3.623.01-.018h4.194a.98.98 0 00.849-.49l2.091-3.623a.98.98 0 000-.98l-2.091-3.623a.98.98 0 00-.849-.49h-4.2zM11.15 19.073l2.091-3.623-2.091-3.623H6.966L4.874 15.45l2.092 3.623h4.184zm9.236-7.754l-2.092 3.623h-4.183l-2.092-3.623 2.092-3.623h4.183l2.092 3.623z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        10 infrastructureText changes
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey100"
        style="display: flex; align-items: center;"
      >
        <span
          class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey100"
          data-icon="warning-outline"
        >
          <svg
            fill="#fff"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.29 3.86L1.82 18a2 2 0 001.71 3h16.94a2 2 0 001.71-3L13.71 3.86a2 2 0 00-3.42 0zM12 9v4M12 17h.01"
            />
          </svg>
        </span>
        10 cv.changeSource.tooltip.incidents
      </p>
    </div>
  </div>
  <div
    class="Layout--vertical Layout--layout-spacing-large StyledProps--main"
  >
    <p
      class="StyledProps--font StyledProps--main serviceHealth StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-tiny-semi"
    >
      cv.monitoredServices.monitoredServiceTabs.serviceHealth
    </p>
    <div
      class="StyledProps--font StyledProps--main StyledProps--padding-top-small StyledProps--padding-bottom-large"
    />
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main healthScoreCardContainer"
    >
      <span
        class="Tag--main"
      >
        <span
          class="bp3-tag healthScoreCard"
        >
          <span
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            45
          </span>
        </span>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-small"
      >
        cv.monitoredServices.serviceHealth.serviceDependencies.states.needsAttention
      </p>
    </div>
    <p
      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-tiny-semi StyledProps--padding-bottom-small"
    >
      cv.dependenciesHealthWithCount
    </p>
    <div
      class="Layout--horizontal StyledProps--main container"
    >
      <span
        class="bp3-popover-wrapper"
      >
        <span
          class="bp3-popover-target"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-green500"
            data-icon="polygon"
            tabindex="0"
          >
            <svg
              height="14"
              viewBox="0 0 9.5 12"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.776.885L9.252 3.41v5.05l-4.476 2.526L.3 8.461v-5.05z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
      </span>
    </div>
    <p
      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-tiny-semi StyledProps--padding-bottom-small"
    >
      cv.slos.title (0)
    </p>
  </div>
</div>
`;
